pyproject.toml

```diff
@@ -3,37 +3,43 @@ name = "mlops-with-databricks"
 version = "0.0.1"
 description = "MLOps with Databricks"
 readme = "README.md"
-requires-python = "==3.11"
-dependencies = ["lightgbm>=4.5.0, <5",
-                "scikit-learn>=1.5.1, <2",
-                "cloudpickle>=3.0.0, <4",
-                "mlflow>=2.16.0, <3",
-                "numpy>=1.26.4, <2",
-                "pandas>=2.2.2, <3",
-                "pyarrow>=15.0.2, <16",
-                "cffi>=1.17.1, <2",
-                "scipy>=1.14.1, <2",
-                "matplotlib>=3.9.2, <4",
-                "databricks-feature-engineering>=0.6, <1"]
+requires-python = "==3.11.10"
+dependencies = [
+    "lightgbm>=4.5.0, <5",
+    "scikit-learn>=1.5.1, <2",
+    "cloudpickle>=3.0.0, <4",
+    "mlflow>=2.16.0, <3",
+    "numpy>=1.26.4, <2",
+    "pandas>=2.2.2, <3",
+    "pyarrow>=15.0.2, <16",
+    "cffi>=1.17.1, <2",
+    "scipy>=1.14.1, <2",
+    "matplotlib>=3.9.2, <4",
+    "databricks-feature-engineering>=0.6, <1",
+    "pre-commit>=4.0.1",
+    "pytest>=8.3.3",
```

Comment:
pytest is not really a project dependency (you do not need it to run code on Databricks), so is pre-commit, those must be in optional dependencies